sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -345,6 +346,66 @@ case class MakeInterval(
     )
 }
 
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = "_FUNC_(years, months) - Make year-month interval from years, months.",
+  arguments = """
+    Arguments:
+      * years - the number of years, positive or negative
+      * months - the number of months, positive or negative
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_(1, 2);
+       1-2
+      > SELECT _FUNC_(1, 0);
+       1-0
+      > SELECT _FUNC_(0, 1);
+       0-1
+  """,
+  since = "3.2.0",
+  group = "datetime_funcs")
+// scalastyle:on line.size.limit
+case class MakeYMInterval(years: Expression, months: Expression)
+  extends BinaryExpression with ImplicitCastInputTypes with NullIntolerant {
+
+  def this(years: Expression) = this(years, Literal(0))
+  def this() = this(Literal(0))
+
+  override def left: Expression = years
+  override def right: Expression = months
+  override def inputTypes: Seq[AbstractDataType] = Seq(IntegerType, IntegerType)
+  override def dataType: DataType = YearMonthIntervalType
+  override def nullable: Boolean = children.exists(_.nullable)
+
+  private def evalIntValue(dt: DataType, value: Any): Long = dt match {
+    case _: ByteType | _: ShortType | _: IntegerType => value.asInstanceOf[Number].longValue()
```

Comment:
If you extend `ImplicitCastInputTypes`, why do you expect other types. Could you explain, please.

Reply:
Seems ansi SQL not support  implicit casts.
![image](https://user-images.githubusercontent.com/46485123/121988862-ad097500-cdcd-11eb-887f-f5ce278cfbad.png)

Reply:
We should handle it like `MultiplyYMInterval` use  `NumericType`?

Reply:
Let's follow `MakeInterval`, and handle/accept only `IntegerType` as it is declared by:
```scala
  override def inputTypes: Seq[AbstractDataType] = Seq(IntegerType, IntegerType)
```

Reply:
ok
 So, should we handle ansi not support implicit casts or it's the standard?